tests/cpp_new/ir/ir_builder_test.cpp

```diff
@@ -44,10 +44,12 @@ TEST(IRBuilder, RangeFor) {
   auto *zero = builder.get_int32(0);
   auto *ten = builder.get_int32(10);
   auto *loop = builder.create_range_for(zero, ten);
-  builder.set_insertion_point_to_loop_begin(loop);
-  auto *index = builder.get_loop_index(loop, 0);
-  builder.set_insertion_point_to_after(loop);
-  auto *ret = builder.create_return(zero);
+  Stmt *index;
+  {
+    auto _ = builder.get_loop_guard(loop);
+    index = builder.get_loop_index(loop, 0);
+  }
+  [[maybe_unused]] auto *ret = builder.create_return(zero);
   EXPECT_EQ(zero->parent->size(), 4);
   ASSERT_TRUE(loop->is<RangeForStmt>());
   auto *loopc = loop->cast<RangeForStmt>();
```

Comment:
Could you add a test that: 

1. create a loop guard,
2. changes the insertion point and insert a few more statements, 
3. verifies that after the loop guard, the insertion point is after the loop